app/controllers/api/v1/specialties_controller.rb

```diff
@@ -0,0 +1,36 @@
+class Api::V1::SpecialtiesController < ApplicationController
+  before_action :set_specialty, only: %i[show destroy]
+
+  def index
+    @specialties = Specialty.all
+    render json: @specialties
+  end
+
+  def show
+    render json: @specialty
+  end
+
+  def create
+    @specialty = Specialty.new(specialty_params)
+
+    if @specialty.save
+      render json: @specialty, status: :created
+    else
+      render json: @specialty.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @specialty.destroy
+  end
```

Comment:
- [ ] What happens if the specialty is not destroyed correctly and the request fails? Kindly add some error handling here. 👍